packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -192,6 +191,19 @@ export const AutocompleteInput = <
 
     const translate = useTranslate();
 
+    const finalChoices =
+        isRequiredOverride || emptyText == null
+            ? allChoices
+            : [
+                  {
+                      id: '',
```

Comment:
This will only work if the options use `id` and `name` as keys. With different options (and usage of `optionText` and `optionValue`, this will fail.